benchmarks/generate_benchmark_params.py

```diff
@@ -0,0 +1,521 @@
+"""Benchmarking parameters generation file."""
+
+import argparse
+from itertools import chain, product
+
+import pandas as pd
+
+parser = argparse.ArgumentParser(
+    description="Generate parameters for which benchmark is run"
+)
+
+parser.add_argument(
+    "-m",
+    "--manifold",
+    type=str,
+    default="all",
+    help="Manifold for which benchmark is run. 'all' denotes all manifolds present.",
+)
+parser.add_argument(
+    "-n",
+    "--n_samples",
+    type=int,
+    default=10,
+    help="Number of samples for which benchmark is run",
+)
+args = parser.parse_args()
+
+
+def spd_manifold_params(n_samples):
+    """Generate spd manifold benchmarking parameters.
+
+    Parameters
+    ----------
+        n_samples : int
+            Number of samples to be used.
+
+    Returns
+    -------
+        _:list.
+        List of params.
+    """
+    manifold = "SPDMatrices"
+    manifold_args = [(2,), (5,), (10,)]
+    module = "geomstats.geometry.spd_matrices"
+
+    def affine_metric_params():
```

Comment:
Need to be `spd_affine_metric_params` for consistency with the rest of this file:

e.g. StiefelCanonicalMetric became stiefel_canonical_metric

The rule to use exactly the name of the metric seems good to me.